### PR TITLE
Update documents, tutorials, and examples.

### PR DESCRIPTION
**Pull request by y0z**
## Contributor Agreements

Please read the [contributor agreements](https://github.com/optuna/optunahub-registry/blob/main/CONTRIBUTING.md#contributor-agreements) and if you agree, please click the checkbox below.

- [x] I agree to the contributor agreements.

## Motivation & Description of the changes

- Add a description about the linter and formatter
- Update tutorials based on the feedback from contributors
- Update the README template to make it more understandable

### CODE REVIEW

File: recipes/002_registration.py [hunk 1]
```diff
@@ -31,12 +32,22 @@
 
 The created directory should include the following files:
 
-- `README.md`: A description of your algorithm. This file is used to create an `web page of OptunaHub <https://hub.optuna.org/>`_. Let me explain the format of the `README.md` file later.
+- `YOUR_ALGORITHM_NAME.py`: The implementation of your algorithm.
 - `__init__.py`: An initialization file. This file must implement your algorithm or import its implementation from another file, e.g., `YOUR_ALGORITHM_NAME.py`.
+- `example.py`: This file should contain a simple example of how to use your algorithm (Example: `example.py for Simulated Annealing Sampler <https://github.com/optuna/optunahub-registry/blob/main/package/samplers/simulated_annealing/example.py>`_).
```
Comment (HideakiImamura) [inline recipes/002_registration.py:37]: It would be great to note that this file is optional.

File: recipes/002_registration.py [hunk 2]
```diff
@@ -14,14 +14,15 @@
 | `package <https://github.com/optuna/optunahub-registry/tree/main/package>`__
 | └── category (e.g. samplers, pruners, and visualization)
 |     └── YOUR_PACKAGE_NAME (you need to create this directory and its contents)
-|         ├── README.md
+|         ├── YOUR_ALGORITHM_NAME.py
 |         ├── __init__.py
+|         ├── example.py
```
Comment (HideakiImamura) [inline recipes/002_registration.py:19]:
```suggestion
|         ├── (example.py)
```

File: template/README.md [hunk 3]
```diff
@@ -1,23 +1,39 @@
...
 If you have additional dependencies, please fill in the installation guide here.
+If no additional dependencies is required, this section can be removed.
 
 ## Example
 Please fill in the code snippet to use the implemented feature here.
 
 ## Others
-Please fill in any other information if you have here.
+Please fill in any other information if you have here by adding child sections (###).
+If there is no additional information, this section can be removed.
```
Comment (HideakiImamura) [inline template/README.md:39]: It would be great to add an example of this section. I think the `Reference` section is an good example.

**Review by HideakiImamura — COMMENTED** [review]
Thank you for the PR. I have several comments. PTAL.

File: template/README.md (hunk 3 above)
Comment (y0z) [reply template/README.md:39]: I've added examples.

**Comment by y0z** [comment]
@HideakiImamura 
Thank you for your feedback. I applied your suggestions. PTAL.

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**